client/src/components/ContentNavigator/ContentDataProvider.ts

```diff
@@ -735,6 +752,30 @@ class ContentDataProvider
         }
       : undefined;
   }
+
+  private async getCachedParentPath(item: ContentItem): Promise<string> {
+    const parentUri = item.parentFolderUri;
+    if (!parentUri) {
+      return "";
+    }
+
+    if (!this.parentPathCache.has(parentUri)) {
+      try {
+        const fullPath = await this.model.getPathOfItem(item);
```

Comment:
It may also be worth updating `RestContentAdapter.getPathOfItem` to handle its own caching since it's really the only `ContentAdapter` that requires it. And doing that w/n RestContentAdapter may make things easier w/r/t Wei's comment about using `uri`